src/k2/persistence/plog_client/PlogClient.cpp

```diff
@@ -169,20 +169,50 @@ seastar::future<std::tuple<Status, uint32_t>> PlogClient::seal(String plogId, ui
     }
 
     return seastar::when_all_succeed(sealFutures.begin(), sealFutures.end())
-        .then([this](std::vector<std::tuple<Status, dto::PlogSealResponse> >&& results) {
+        .then([this](std::vector<std::tuple<Status, dto::PlogSealResponse> >&& results) { 
             Status return_status;
             uint32_t sealed_offset;
             for (auto& result: results){
                 auto& [status, response] = result;
                 return_status = std::move(status);
                 sealed_offset = response.sealedOffset;
-                if (!return_status.is2xxOK())
+                if (!return_status.is2xxOK()) 
                     break;
             }
             return seastar::make_ready_future<std::tuple<Status, uint32_t> >(std::tuple<Status, uint32_t>(std::move(return_status), std::move(sealed_offset)));
         });
 }
 
+seastar::future<std::tuple<Status, std::tuple<uint32_t, bool>>> PlogClient::getPlogStatus(String plogId){
+    dto::PlogStatusRequest request{.plogId = std::move(plogId)};
+
+    std::vector<seastar::future<std::tuple<Status, dto::PlogStatusResponse> > > statusFutures;
+    for (auto& ep:_persistenceMapEndpoints[_persistenceNameList[_persistenceMapPointer]]){
+        statusFutures.push_back(RPC().callRPC<dto::PlogStatusRequest, dto::PlogStatusResponse>(dto::Verbs::PERSISTENT_STATUS, request, *ep, _plog_timeout()));
+    }
+
+    return seastar::when_all_succeed(statusFutures.begin(), statusFutures.end())
+        .then([this](std::vector<std::tuple<Status, dto::PlogStatusResponse> >&& results) { 
+            Status return_status;
+            uint32_t current_offset=UINT_MAX;
+            bool sealed=false;
+            for (auto& result: results){
+                auto& [status, response] = result;
+                return_status = std::move(status);
+                if (current_offset > response.currentOffset){
+                    current_offset = response.currentOffset;
+                }
+                if (sealed < response.sealed){

```

Comment:
sealed is a bool, is that what you what here?

Reply:
changed